### PR TITLE
DM-31494:  Use UTC throughout

### PR DESCRIPTION
**Pull request by SimonKrughoff**

### CODE REVIEW

**Comment by SimonKrughoff** [comment]
This can't be merged until we fix the issue with the resample function in DM-28462

**Comment by SimonKrughoff** [comment]
DM-28462 has merged and this branch was rebased on it

File: lsst_efd_client/efd_helper.py [hunk 1]
```diff
@@ -26,15 +26,17 @@ class EfdClient:
...
         (``https://roundtable.lsst.codes/segwarides/`` by default).
     client : `object`, optional
         An instance of a class that ducktypes as `aioinflux.InfluxDBClient`.
         The intent is to be able to substitute a mocked client for testing.
+    convert_influx_index : `bool`, optional
+        Convert influxDB time index from TAI to UTC?  This is for using legacy
+        instances that may still have timestamps stored internally as TAI.
+        Modern instances all store index timestamps as UTC natively.
+        Default is `False`, don't translate from TAI to UTC.
     """
```
Comment (afausti) [inline lsst_efd_client/efd_helper.py:40]: I like this option.

File: lsst_efd_client/efd_helper.py [hunk 2]
```diff
@@ -205,7 +212,8 @@ def build_time_range_query(self, topic_name, fields, start, end, is_window=False
                 start_str = start.isot
                 end_str = (start + end).isot
         elif isinstance(end, Time):
-            end = getattr(end, self.internal_scale)
+            if not end.scale == 'utc':
+                raise ValueError(f'Timestamps must be in UTC.')
```
Comment (afausti) [inline lsst_efd_client/efd_helper.py:216]: I see that you decided not to convert the `start` and `end` time scale anymore, and defer it to the user instead. I think it is fine.

**Review by afausti — APPROVED** [review]
Looks great, thanks for doing that.

File: lsst_efd_client/efd_helper.py (hunk 2 above)
Comment (SimonKrughoff) [reply lsst_efd_client/efd_helper.py:216]: Yes.  I thought about issuing a warning as we did before, but now that we only allow one internal scale, it actually seemed more dangerous to do that than to except early and make the user do the conversion explicitly

**Pull request merged**